### ISSUE TITLE
Triton installation fails

### ISSUE THREAD

**Issue by andrewmogan** [FNALssi/fnal_art#77]
`spack install triton` fails when adding `fnal_art` into a fresh spack environment. Steps to reproduce the issue:
```
git clone --depth=100 --branch=releases/v0.22 https://github.com/spack/spack.git ~/spack
cd ~/spack
 . share/spack/setup-env.sh
cd ~
git clone https://github.com/FNALssi/fnal_art.git
spack repo add fnal_art
spack info triton # Works
spack install triton # Spins its wheels for a while before failing 
```

I ran this as myself on `daq.fnal.gov`, which has been upgraded to AL9. The last command fails when building `grpc`. The full error output is very long, but I think the relevant bit is

```
  >> 927     /tmp/amogan/spack-stage/spack-stage-grpc-1.55.0-l7bz3k2qzquzgwlhjxzgmix6dbek7f2t/spack-src/src/core/lib/iomgr/event_engine_shims/tcp_client.cc:74:23: e
             rror: 'StrCat' is not a member of 'absl'
     928        74 |                 absl::StrCat("tcp-client:", addr_uri.value()))
     929           |                       ^~~~~~
```

After a bunch of Googling, it seems like `abseil` and `grpc` have a troubled history of not talking to each other and breaking things. [This abseil issue](https://github.com/abseil/abseil-cpp/issues/1624) seems to suggest some mutex destructor is missing, but I'm not sure if that's what's causing this problem. 

In any case, it seems like there's some incompatibility between `grpc` and `abseil` in this environment. I've tried a handful of different versions of each with similar results. I also tried compiling with `gcc@13.2.0`, which failed with a different error that still appeared to be related to a conflict between these two packages. 

Any suggestions would be much appreciated.

**Comment by marcmengel** [comment]
We haven't updated versions on abseil-cpp in a while, mabye do a:

spack checksum abseil-cpp  20240116.2

add the resulting version line to the recipe with 'spack edit abseil-cpp' and try and build that version? 
It may actually be an incompatability with, i.e. gcc@13 which is already fixed.   I think the last time
the whole triton thing was built successfully it was with gcc@8.2.0 or so...

**Comment by marcmengel** [comment]
Or alternately, specify that we want gcc@8 for triton and abseil-cpp?

**Comment by andrewmogan** [comment]
Thanks, Marc.

Unfortunately, I get similar errors in both cases. For example, when building with gcc@8.2.0, 

```
  >> 940     /tmp/amogan/spack-stage/spack-stage-grpc-1.55.0-w2g4l4uxwehlezbvp76iwomwhojrzeyq/spack-src/src/core/lib/iomgr/event_engine_shims/tcp_client.cc:74:23: er
             ror: 'StrCat' is not a member of 'absl'
     941        74 |                 absl::StrCat("tcp-client:", addr_uri.value()))
     942           |    
```

Adding the newer checksum for `abseil-cpp 20240116.2` also had the same issue.

**Comment by marcmengel** [comment]
Did it actually build the latest abseil-cpp after you added the version?   If so we may need to actually patch that tcp_client.cc 
in the recipe; perhaps to add an ` #include <absl/strings/str_cat.h>` ?

**Comment by andrewmogan** [comment]
Yes, I can see this line in the output during the install:
```
Installing abseil-cpp-20240116.2-ti5bd3iedsluao235ibmmjilnvtea7xh [35/51]
```

However, it seems like upgrading to `grpc@1.61.0` along with that newer `abseil-cpp` works! Other versions may work as well; I just tried that one on a whim.

**Comment by marcmengel** [comment]
If you get that part of the build stack working, send a spack spec -I of it along to
the list, so we can update required versions for upcoming builds.

--
Marc W. Mengel
Computer Services Senior Developer

Data Management and Applications
Fermi National Accelerator Laboratory
630 840 8256 office
www.fnal.gov<https://email.fnal.gov/owa/redir.aspx?C=JkVsWjSyHEuWV2kFJWNPyC9OyMKf7dEIp_yxPk9yfhfadE3mdLfI6T8Ztf8BVaJjzoIe8oJ5H0M.&URL=http%3a%2f%2fwww.fnal.gov>
<https://email.fnal.gov/owa/redir.aspx?C=JkVsWjSyHEuWV2kFJWNPyC9OyMKf7dEIp_yxPk9yfhfadE3mdLfI6T8Ztf8BVaJjzoIe8oJ5H0M.&URL=http%3a%2f%2fwww.fnal.gov>
Connect with us!
Newsletter<https://email.fnal.gov/owa/redir.aspx?C=-29AtpIe40SAvGF4aiNr86KvVjap7dEIuxIdg8Q5bYa9sT7QTFVhxhVnl8zwXkEuBSuAQLRkEfg.&URL=http%3a%2f%2fwww.fnal.gov%2fpub%2ftoday> | Facebook<https://email.fnal.gov/owa/UrlBlockedError.aspx> | Twitter<https://email.fnal.gov/owa/redir.aspx?C=-29AtpIe40SAvGF4aiNr86KvVjap7dEIuxIdg8Q5bYa9sT7QTFVhxhVnl8zwXkEuBSuAQLRkEfg.&URL=https%3a%2f%2ftwitter.com%2fFermilabToday>
________________________________

**Comment by andrewmogan** [comment]
Sure thing. Which list, specifically?

**Comment by marcmengel** [comment]
Actually, just posting it here on the ticket should be fine.

**Comment by andrewmogan** [comment]
Here, I'm using `gcc@11.4.1` because that's what came out of the box when I setup spack. But the build also seems to work with `gcc@8.2.0` and `gcc@13.2.0`. 

TL;DR I updated to `abseil-cpp@20240116.2` and `grpc@1.61.0`. There are newer versions of `grpc` that may work as well, but I haven't tried them. 
```
amogan@daq:~/spack$ spack spec -l triton
Input spec
--------------------------------
 -   triton

Concretized
--------------------------------
[+]  tub3v6e  triton@23.09%gcc@11.4.1~cuda~ipo build_system=cmake build_type=Release generator=make arch=linux-almalinux9-zen3
[+]  ypbwh7u      ^abseil-cpp@20240116.2%gcc@11.4.1~ipo+shared build_system=cmake build_type=Release cxxstd=14 generator=make arch=linux-almalinux9-zen3
[+]  bn2ebyy      ^c-ares@1.27.0%gcc@11.4.1~ipo build_system=cmake build_type=Release generator=make arch=linux-almalinux9-zen3
[+]  q7jw434      ^cmake@3.27.9%gcc@11.4.1~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-almalinux9-zen3
[+]  dewgz2x          ^ncurses@6.5%gcc@11.4.1~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-almalinux9-zen3
[+]  uzzfyl2          ^zlib-ng@2.1.6%gcc@11.4.1+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-almalinux9-zen3
[+]  gytwb4o      ^curl@8.7.1%gcc@11.4.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-almalinux9-zen3
[+]  ob2ebbs          ^nghttp2@1.57.0%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  7bmwc7t              ^diffutils@3.10%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  fhlosnb          ^perl@5.38.0%gcc@11.4.1+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-almalinux9-zen3
[+]  cepu74p              ^berkeley-db@18.1.40%gcc@11.4.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-almalinux9-zen3
[+]  xbdphso              ^bzip2@1.0.8%gcc@11.4.1~debug~pic+shared build_system=generic arch=linux-almalinux9-zen3
[+]  qz6gbyw              ^gdbm@1.23%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  6yi6wxc          ^pkgconf@2.2.0%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  i5myjny      ^gcc-runtime@11.4.1%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[e]  vr7gip6      ^glibc@2.34%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  ppzyjdw      ^gmake@4.4.1%gcc@11.4.1~guile build_system=generic arch=linux-almalinux9-zen3
[+]  pc53sah      ^googletest@1.12.1%gcc@11.4.1+gmock~ipo+pthreads+shared build_system=cmake build_type=Release cxxstd=11 generator=make arch=linux-almalinux9-zen3
[+]  7njb7c5      ^grpc@1.61.0%gcc@11.4.1+codegen~ipo~shared build_system=cmake build_type=Release cxxstd=11 generator=make arch=linux-almalinux9-zen3
[+]  nb2f4kr      ^openssl@3.3.0%gcc@11.4.1~docs+shared build_system=generic certs=mozilla arch=linux-almalinux9-zen3
[+]  kurk3aw          ^ca-certificates-mozilla@2023-05-30%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[+]  6liflwt      ^patchelf@0.17.2%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  ue7j7a6      ^protobuf@3.25.3%gcc@11.4.1~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-almalinux9-zen3
[+]  2ezcilf      ^py-geventhttpclient@1.4.4%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  l7ne5oh          ^py-pip@23.1.2%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[+]  eatcgy6          ^python@3.11.7%gcc@11.4.1+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-almalinux9-zen3
[+]  eb5sjm5              ^expat@2.6.2%gcc@11.4.1+libbsd build_system=autotools arch=linux-almalinux9-zen3
[+]  rsl3rjx                  ^libbsd@0.12.1%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  7jucked                      ^libmd@1.0.4%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  pqffw4d              ^gettext@0.22.5%gcc@11.4.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-almalinux9-zen3
[+]  fjilbzl                  ^libxml2@2.10.3%gcc@11.4.1+pic~python+shared build_system=autotools arch=linux-almalinux9-zen3
[+]  2zi7fmg                  ^tar@1.34%gcc@11.4.1 build_system=autotools zip=pigz arch=linux-almalinux9-zen3
[+]  vky7umw                      ^pigz@2.8%gcc@11.4.1 build_system=makefile arch=linux-almalinux9-zen3
[+]  pu3yf6l                      ^zstd@1.5.6%gcc@11.4.1+programs build_system=makefile compression=none libs=shared,static arch=linux-almalinux9-zen3
[+]  wrb2o2m              ^libffi@3.4.6%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  f4rrlhs              ^libxcrypt@4.4.35%gcc@11.4.1~obsolete_api build_system=autotools patches=4885da3 arch=linux-almalinux9-zen3
[+]  lgvya55              ^readline@8.2%gcc@11.4.1 build_system=autotools patches=bbf97f1 arch=linux-almalinux9-zen3
[+]  xrthtyh              ^sqlite@3.43.2%gcc@11.4.1+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-almalinux9-zen3
[+]  7ijppn7              ^util-linux-uuid@2.38.1%gcc@11.4.1 build_system=autotools arch=linux-almalinux9-zen3
[+]  jcrslac              ^xz@5.4.6%gcc@11.4.1~pic build_system=autotools libs=shared,static arch=linux-almalinux9-zen3
[+]  dbaek5i          ^python-venv@1.0%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[+]  pgz25mu      ^py-grpcio@1.60.1%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  rqgncoj          ^py-cython@0.29.36%gcc@11.4.1 build_system=python_pip patches=c4369ad arch=linux-almalinux9-zen3
[+]  4jarczr      ^py-grpcio-tools@1.59.0%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  m3rqh2k      ^py-numpy@1.26.4%gcc@11.4.1 build_system=python_pip patches=873745d arch=linux-almalinux9-zen3
[+]  uhyftgn          ^openblas@0.3.26%gcc@11.4.1~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=linux-almalinux9-zen3
[+]  6lst4ie          ^py-meson-python@0.15.0%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  xkxbzyd              ^meson@1.3.2%gcc@11.4.1 build_system=python_pip patches=0f0b1bd arch=linux-almalinux9-zen3
[+]  374pfmf                  ^ninja@1.11.1%gcc@11.4.1+re2c build_system=generic arch=linux-almalinux9-zen3
[+]  bgegqqw                      ^re2c@2.2%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[+]  lxpenva              ^py-pyproject-metadata@0.7.1%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  ddh34em                  ^py-packaging@23.1%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  kg373o4                      ^py-flit-core@3.9.0%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  m4ayb46      ^py-python-rapidjson@1.10%gcc@11.4.1 build_system=python_pip arch=linux-almalinux9-zen3
[+]  vvkx57x      ^py-setuptools@69.2.0%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[+]  w4xmzc5      ^py-wheel@0.41.2%gcc@11.4.1 build_system=generic arch=linux-almalinux9-zen3
[+]  xhljbc2      ^rapidjson@1.2.0-2022-03-09%gcc@11.4.1~doc~ipo build_system=cmake build_type=Release generator=make patches=a4316c5 arch=linux-almalinux9-zen3
[+]  yzphmun      ^re2@2023-09-01%gcc@11.4.1~ipo+pic+shared build_system=cmake build_type=Release generator=make arch=linux-almalinux9-zen3
```
